django/contrib/admin/views/autocomplete.py

```diff
@@ -24,12 +25,19 @@ def get(self, request, *args, **kwargs):
             return JsonResponse({'error': '403 Forbidden'}, status=403)
 
         self.term = request.GET.get('term', '')
-        self.paginator_class = self.model_admin.paginator
+        app_label = request.GET.get('app_label')
+        model_name = request.GET.get('model_name')
```

Comment:
Are these really necessary? Can't they be retrieved from `self.object_list.model`?

Reply:
No, this is not. they are different models. Once being the source with the FK, the other the model the FK points to.

Reply:
I see. Any thoughts about passing the _local_ field instead of the remote one and the model it's from?

e.g. in the case of tests below `field_name` would be `'uuid'` instead of `'question_with_to_field'`.

That's how the other `to_field` enabled admin views work and what `to_field_allowed` expects. It also feels like a nicer API to me and would allow you to stop passing `app_label` and `model_name`.

Reply:
@charettes the `field_name` would be `question_with_to_field` os the local field. See the new tests cases I added.

Reply:
In any case, naming could be improved I guess to avoid the confusion.

Reply:
@codingjoe it's the remote field in test and not the local one.

The URL is `'autocomplete_admin:admin_views_question_autocomplete'` (which is bound to the `Question` model) and `'question_with_to_field'` is an `Answer` field.

What I'm suggesting we do instead is pass `field='uuid'` which is a field local to `Question` and have `Answer` admin pass `field.remote_field.name` instead which is `'uuid'`.

That's what the other admin views are doing and doesn't require you do extra `model` and `app_label` sanitization when walking your way back to the local field from the remote one.

Reply:
I agree with Simon here, if you for instance look at the edit popup next to foreignkeys you will see this in the URL: `http://127.0.0.1:8000/admin/auth/user/apo/change/?_to_field=username&_popup=1`

So I propose that the autocomplete request should look like this: `http://127.0.0.1:8000/admin/auth/user/autocomplete/?_to_field=username`

and not like currently: `http://127.0.0.1:8000/admin/auth/user/autocomplete/?app_label=select2&model_name=test&field_name=user`

Reply:
As much as I want to agree with the two of you, I can't. The solution you are proposing only solves the `to_field` issue #29138 but not #29010 which is about `limit_choices_to`. For that to work I need to local field and not only the `to_field`.

To be honest here, the `to_field` issues, is somewhat of an edge case, I see the `limit_choices_to` problem as much more severe, since it's a more commonly used feature.